SimTKcommon/include/SimTKcommon/internal/Xml.h

```diff
@@ -1708,10 +1709,76 @@ template <class T> inline
 auto fromXmlElementHelper(T& thing, Xml::Element& elt, 
                           const std::string& requiredTag, int) 
     -> void {
+    // TODO std::cout << "DEBUGfromXmlfree" << elt << std::endl;
     fromXmlElement(thing, elt, requiredTag); // call free function
 }
 
 
+inline void fromXmlElementHelperHelperRecurse(Xml::element_iterator eit,
+        Xml::element_iterator end) {
+    assert(eit == end);
+}
+
+/*
+// TODO decide between this and the other interface. 
+// With this interface, the call
+// is more verbose but also more clear:
+//
+// fromXmlElementHelperHelper("Foo", 1, e, "reqName",
+//     std::make_pair{bar, "bar"},
+//     std::make_pair{baz, "baz"}
+//     );
+template <typename T, typename ... TArgs> inline
+void fromXmlElementHelperHelperRecurse(Xml::element_iterator eit,
+        std::pair<T&, std::string> firstField,
+        std::pair<TArgs&, std::string> ... fields) {
```

Comment:
Pointers would be fine here I think. That would mean we'd write `std::make_pair<&thing, "name">` right? That looks good to me now. I've gotten a little more relaxed about pointers for output variables since it is the Google style that we're using in Drake. I can see the appeal somewhat  since the `&thing` makes it clear that the variable is going to be modified. I *think* that is still nicer than separate arguments -- what do you think?